Docm verify option page is displayed after choosing items from section category and section style

KRJuuga opened the pull request:
- Create folder WhatMyCarWorthPage in KBBMOOCProject folder in pages folder
- Create file CategoryAndStylePage.java in WhatMyCarWorthPage folder
- Create file MyCarValuePage.java in WhatMyCarWorthPage folder
- Create file OptionPage.java in WhatMyCarWorthPage folder
- Create folder WhatMyCarWorthTest in KBBMOOCProject folder in tests folder
- Create file CategoryAndStylePageTest.java in WhatMyCarWorthTest
- Delete file phpAccount.java in datadriven folder
- Create file dataJson.java in datadriven folder
- Delete SauceLabsFinalPages folder
- Delete SauceLabsFinalTests folder
- Delete target folder
- Change url in configs.properties to https://staging.kbb.com/whats-my-car-worth/
- Change keyword in file CustomKeyword.java from click to clickThenWait
- Change chromedriver due to different version and system
- Increase wait time to 5s 
- Move Assert function to OptionPage

KRJuuga committed:
Add files via upload

- Create folder WhatMyCarWorthPage in KBBMOOCProject folder in pages folder
- Create file CategoryAndStylePage.java in WhatMyCarWorthPage folder
- Create file MyCarValuePage.java in WhatMyCarWorthPage folder
- Create file OptionPage.java in WhatMyCarWorthPage folder
- Create folder WhatMyCarWorthTest in KBBMOOCProject folder in tests folder
- Create file CategoryAndStylePageTest.java in WhatMyCarWorthTest
- Delete file phpAccount.java in datadriven folder
- Create file dataJson.java in datadriven folder
- Delete SauceLabsFinalPages folder
- Delete SauceLabsFinalTests folder
- Delete target folder
- Change url in configs.properties to https://staging.kbb.com/whats-my-car-worth/
- Change keyword in file CustomKeyword.java from click to clickThenWait
- Change chromedriver due to different version and system

KRJuuga committed:
Delete hcm-aut04 directory

KRJuuga committed:
Add files via upload

- Create folder WhatMyCarWorthPage in KBBMOOCProject folder in pages folder
- Create file CategoryAndStylePage.java in WhatMyCarWorthPage folder
- Create file MyCarValuePage.java in WhatMyCarWorthPage folder
- Create file OptionPage.java in WhatMyCarWorthPage folder
- Create folder WhatMyCarWorthTest in KBBMOOCProject folder in tests folder
- Create file CategoryAndStylePageTest.java in WhatMyCarWorthTest
- Delete file phpAccount.java in datadriven folder
- Create file dataJson.java in datadriven folder
- Delete SauceLabsFinalPages folder
- Delete SauceLabsFinalTests folder
- Delete target folder
- Change url in configs.properties to https://staging.kbb.com/whats-my-car-worth/
- Change keyword in file CustomKeyword.java from click to clickThenWait
- Change chromedriver due to different version and system

KRJuuga committed:
Delete target folder

- Create folder WhatMyCarWorthPage in KBBMOOCProject folder in pages folder
- Create file CategoryAndStylePage.java in WhatMyCarWorthPage folder
- Create file MyCarValuePage.java in WhatMyCarWorthPage folder
- Create file OptionPage.java in WhatMyCarWorthPage folder
- Create folder WhatMyCarWorthTest in KBBMOOCProject folder in tests folder
- Create file CategoryAndStylePageTest.java in WhatMyCarWorthTest
- Delete file phpAccount.java in datadriven folder
- Create file dataJson.java in datadriven folder
- Delete SauceLabsFinalPages folder
- Delete SauceLabsFinalTests folder
- Change url in configs.properties to https://staging.kbb.com/whats-my-car-worth/
- Change keyword in file CustomKeyword.java from click to clickThenWait
- Change chromedriver due to different version and system

KRJuuga committed:
- Create folder WhatMyCarWorthPage in KBBMOOCProject folder in pages folder - Create file CategoryAndStylePage.java in WhatMyCarWorthPage folder - Create file MyCarValuePage.java in WhatMyCarWorthPage folder - Create file OptionPage.java in WhatMyCarWorthPage folder - Create folder WhatMyCarWorthTest in KBBMOOCProject folder in tests folder - Create file CategoryAndStylePageTest.java in WhatMyCarWorthTest - Delete file phpAccount.java in datadriven folder - Create file dataJson.java in datadriven folder - Delete SauceLabsFinalPages folder - Delete SauceLabsFinalTests folder - Delete target folder - Change url in configs.properties to https://staging.kbb.com/whats-my-car-worth/ - Change keyword in file CustomKeyword.java from click to clickThenWait - Change chromedriver due to different version and system

KRJuuga committed:
- Delete SauceLabsFinalTests folder

KRJuuga committed:
- Delete SauceLabsFinalPages folder

KRJuuga committed:
- Create file dataJson.java in datadriven folder

KRJuuga committed:
- Create folder WhatMyCarWorthPage in KBBMOOCProject folder in pages folder - Create file CategoryAndStylePage.java in WhatMyCarWorthPage folder - Create file MyCarValuePage.java in WhatMyCarWorthPage folder - Create file OptionPage.java in WhatMyCarWorthPage folder - Create folder WhatMyCarWorthTest in KBBMOOCProject folder in tests folder - Create file CategoryAndStylePageTest.java in WhatMyCarWorthTest - Delete file phpAccount.java in datadriven folder - Create file dataJson.java in datadriven folder - Delete SauceLabsFinalPages folder - Delete SauceLabsFinalTests folder - Delete target folder - Change url in configs.properties to https://staging.kbb.com/whats-my-car-worth/ - Change keyword in file CustomKeyword.java from click to clickThenWait - Change chromedriver due to different version and system

KRJuuga committed:
Add Assertion in file CategoryAndStylePageTest.java in WhatMyCarWorthTest

- Create folder WhatMyCarWorthPage in KBBMOOCProject folder in pages folder
- Create file CategoryAndStylePage.java in WhatMyCarWorthPage folder
- Create file MyCarValuePage.java in WhatMyCarWorthPage folder
- Create file OptionPage.java in WhatMyCarWorthPage folder
- Create folder WhatMyCarWorthTest in KBBMOOCProject folder in tests folder
- Create file CategoryAndStylePageTest.java in WhatMyCarWorthTest
- Delete file phpAccount.java in datadriven folder
- Create file dataJson.java in datadriven folder
- Delete SauceLabsFinalPages folder
- Delete SauceLabsFinalTests folder
- Delete target folder
- Change url in configs.properties to https://staging.kbb.com/whats-my-car-worth/
- Change keyword in file CustomKeyword.java from click to clickThenWait
- Change chromedriver due to different version and system
- Add Assertion in file CategoryAndStylePageTest.java in WhatMyCarWorthTest

namnguyen-itvn commented:
@chiennv2042 pls review this RP

namnguyen-itvn commented:
@Datvh1422 pls review this PR

chiennv2042 commented:
Review this testcase: 

- Coding convention -> fine
- Verify title should handle in pages folder despite tests folder (using function boolean)
- No stable, sleep() in function clickThenWait() is 2 second, sometimes no long enough to wait the website load other pages.

Datvh1422 commented:
Coding convention: Fine
verify option page is displayed after choosing items from section category and section style
Testrun Ok
Stable, Testcase name ok, Have Function Assert to verify TestCase.

KRJuuga committed:
Increase wait time to 5s - Move Assert function to OptionPage

chiennv2042 commented on demo/src/test/java/com/example/test/core/configs.properties:
```diff
@@ -1,4 +1,4 @@
 browser=CHROME
-url=https://www.saucedemo.com/
+url=https://staging.kbb.com/whats-my-car-worth/
```
url should be effect for all test, all team. should change to specific name

chiennv2042 replied:
url should be effect for all test, all team. should change to specific name

chiennv2042 commented on demo/src/test/java/com/example/test/core/datadriven/dataJson.java:
```diff
@@ -0,0 +1,13 @@
+public class dataJson {
+    @DataProvider(name = "dataCategoryAndStyle")
+    public static Object dataCategoryAndStyle () {
+        return JsonHelper.ReadJsonFromFile("src//test//java//com//example//test//core//resources//datas//datas.json");       
```
should using the most popular root path of file

chiennv2042 commented on demo/src/test/java/com/example/test/core/helper/ReadConfigPropertyFile.java:
```diff
@@ -5,7 +5,7 @@
 public class ReadConfigPropertyFile {
 
-    public static String propertiesFilePath ="src\\test\\java\\com\\example\\test\\core\\configs.properties";
+    public static String propertiesFilePath ="src//test//java//com//example//test//core//configs.properties";
```
should using the most popular root path of file

chiennv2042 commented on demo/src/test/java/com/example/test/core/keywork/CustomKeyword.java:
```diff
@@ -1,151 +1,94 @@
+    public void sendKeys(WebElement element, String text) throws InterruptedException {
+        try {
+            waitForElementDisplayed(element).sendKeys(text);
+            Thread.sleep(2000);
```
should remove thread.sleep and find the way to handle timeout

chiennv2042 commented on demo/src/test/java/com/example/test/core/keywork/CustomKeyword.java:
```diff
@@ -1,151 +1,94 @@
+        try {
+            waitForElementDisplayed(element);
+            element.click();
+            Thread.sleep(5000);
```
should remove thread.sleep and find the way to handle timeout

chiennv2042 commented on demo/src/test/java/com/example/test/pages/KBBMOOCProject/WhatMyCarWorthPage/CategoryAndStylePage.java:
```diff
@@ -0,0 +1,54 @@
+    }
+
+    public void clickToChooseAstyleInStyleSectionInCategoryAndStylePage() throws InterruptedException {
+        WebElement btnStyle = keyword.findWebElementByXpath("//div[text()='230i xDrive Coupe 2D']");
```
should write the web element locator in separate line then call into a method/function

chiennv2042 commented on demo/src/test/java/com/example/test/pages/KBBMOOCProject/WhatMyCarWorthPage/CategoryAndStylePage.java:
```diff
@@ -0,0 +1,54 @@
+    }
+
+    public void clickOnNextButtonInCategoryAndStylePage() throws InterruptedException {
+        WebElement btnNext = keyword.findWebElementByXpath("//span[text()='Next']");
```
should write the web element locator in separate line then call into a method/function

chiennv2042 commented on demo/src/test/java/com/example/test/pages/KBBMOOCProject/WhatMyCarWorthPage/CategoryAndStylePage.java:
```diff
@@ -0,0 +1,54 @@
+    }
+
+    public void clickOnStyleDropdownInCategoryAndStylePage() throws InterruptedException {
+        WebElement btnStyle = keyword.findWebElementByXpath("//div[@id='style']");
```
should write the web element locator in separate line then call into a method/function

chiennv2042 commented on demo/src/test/java/com/example/test/pages/KBBMOOCProject/WhatMyCarWorthPage/CategoryAndStylePage.java:
```diff
@@ -0,0 +1,54 @@
+    }
+
+    public void clickToChooseaStyleinStyleDropdownInCategoryAndStylePage() throws InterruptedException {
+        WebElement btnChoose = keyword.findWebElementByXpath("//div[@id='style']//option[@value='434546']");
```
should write the web element locator in separate line then call into a method/function

chiennv2042 commented on demo/src/test/java/com/example/test/pages/KBBMOOCProject/WhatMyCarWorthPage/CategoryAndStylePage.java:
```diff
@@ -0,0 +1,54 @@
+    }
+
+    public void sendKeysToMileageTextBoxInCategoryAndStylePage(String mileage) throws InterruptedException {
+        WebElement inputMileage = keyword.findWebElementByXpath("//input[@data-lean-auto='mileageInput']");
```
should write the web element locator in separate line then call into a method/function

chiennv2042 commented on demo/src/test/java/com/example/test/pages/KBBMOOCProject/WhatMyCarWorthPage/CategoryAndStylePage.java:
```diff
@@ -0,0 +1,54 @@
+    }
+
+    public void sendKeysToZipTextBoxInCategoryAndStylePage(String zip) throws InterruptedException {
+        WebElement inputZip = keyword.findWebElementByXpath("//input[@data-lean-auto='zipcodeInput']");
```
should write the web element locator in separate line then call into a method/function

chiennv2042 commented on demo/src/test/java/com/example/test/pages/KBBMOOCProject/WhatMyCarWorthPage/MyCarValuePage.java:
```diff
@@ -0,0 +1,86 @@
+   }
+
+   public void clickOnYearDropDownInMyCarValuePage() throws InterruptedException {
+      WebElement btnYear = keyword.findWebElementByXpath("//div[@class='year']");
```
should write the web element locator in separate line then call into a method/function

chiennv2042 commented on demo/src/test/java/com/example/test/pages/KBBMOOCProject/WhatMyCarWorthPage/MyCarValuePage.java:
```diff
@@ -0,0 +1,86 @@
+   }
+
+   public void clickToChooseaYearInYearDropDownInMyCarValuePage() throws InterruptedException {
+      WebElement optYear = keyword.findWebElementByXpath("//div[@class='year']//option[@value= '2020']");
```
should write the web element locator in separate line then call into a method/function

chiennv2042 commented on demo/src/test/java/com/example/test/pages/KBBMOOCProject/WhatMyCarWorthPage/MyCarValuePage.java:
```diff
@@ -0,0 +1,86 @@
+   }
+
+   public void clickOnMakeDropDownInMyCarValuePage() throws InterruptedException {
+      WebElement btnMake = keyword.findWebElementByXpath("//div[@class='make']");
```
should write the web element locator in separate line then call into a method/function

chiennv2042 commented on demo/src/test/java/com/example/test/pages/KBBMOOCProject/WhatMyCarWorthPage/MyCarValuePage.java:
```diff
@@ -0,0 +1,86 @@
+   }
+
+   public void clickToChooseaMakeInMakeDropDownInMyCarValuePage() throws InterruptedException {
+      WebElement optMake = keyword.findWebElementByXpath("//div[@class='make']//option[@value= '5']");
```
should write the web element locator in separate line then call into a method/function

chiennv2042 commented on demo/src/test/java/com/example/test/pages/KBBMOOCProject/WhatMyCarWorthPage/MyCarValuePage.java:
```diff
@@ -0,0 +1,86 @@
+   }
+
+   public void clickOnModelDropDownInMyCarValuePage() throws InterruptedException {
+      WebElement btnModel = keyword.findWebElementByXpath("//div[@class='model']");
```
should write the web element locator in separate line then call into a method/function

chiennv2042 commented on demo/src/test/java/com/example/test/pages/KBBMOOCProject/WhatMyCarWorthPage/MyCarValuePage.java:
```diff
@@ -0,0 +1,86 @@
+   }
+
+   public void clickToChooseaModelInModelDropDownInMyCarValuePage() throws InterruptedException {
+      WebElement optModel = keyword.findWebElementByXpath("//div[@class='model']//option[@value= '32913']");
```
should write the web element locator in separate line then call into a method/function

chiennv2042 commented on demo/src/test/java/com/example/test/pages/KBBMOOCProject/WhatMyCarWorthPage/MyCarValuePage.java:
```diff
@@ -0,0 +1,86 @@
+   }
+
+   public void sendKeysToMileageTextBoxInMyCarValuePage(String mileage) throws InterruptedException {
+      WebElement inputMileage = keyword
```
should write the web element locator in separate line then call into a method/function

chiennv2042 commented on demo/src/test/java/com/example/test/pages/KBBMOOCProject/WhatMyCarWorthPage/MyCarValuePage.java:
```diff
@@ -0,0 +1,86 @@
+   }
+
+   public void sendKeysToZipTextBoxInMyCarValuePage(String zip) throws InterruptedException {
+      WebElement inputZip = keyword
```
should write the web element locator in separate line then call into a method/function

chiennv2042 commented on demo/src/test/java/com/example/test/pages/KBBMOOCProject/WhatMyCarWorthPage/MyCarValuePage.java:
```diff
@@ -0,0 +1,86 @@
+   }
+
+   public void clickOnGoButtonInMyCarValuePage() throws InterruptedException {
+      WebElement btnGo = keyword.findWebElementByXpath("//span[text()='Go']");
```
should write the web element locator in separate line then call into a method/function

chiennv2042 commented on demo/src/test/java/com/example/test/pages/KBBMOOCProject/WhatMyCarWorthPage/MyCarValuePage.java:
```diff
@@ -0,0 +1,86 @@
+   }
+
+   public void sendKeysToPlateTextBoxInMyCarValuePage(String plate) throws InterruptedException {
+      WebElement inputPlate = keyword.findWebElementByXpath("//div[@id='lpNumberInput']//input");
```
should write the web element locator in separate line then call into a method/function

chiennv2042 commented on demo/src/test/java/com/example/test/pages/KBBMOOCProject/WhatMyCarWorthPage/MyCarValuePage.java:
```diff
@@ -0,0 +1,86 @@
+      keyword.sendKeys(inputPlate, plate);
+   }
+   public void clickOnStateDropDownInMyCarValuePage() throws InterruptedException {
+      WebElement btnState = keyword.findWebElementByXpath("//div[@id='state']");
```
should write the web element locator in separate line then call into a method/function

chiennv2042 commented on demo/src/test/java/com/example/test/pages/KBBMOOCProject/WhatMyCarWorthPage/MyCarValuePage.java:
```diff
@@ -0,0 +1,86 @@
+   }
+
+   public void clickToChooseStateInStateDropDownInMyCarValuePage() throws InterruptedException {
+      WebElement optState = keyword.findWebElementByXpath("//div[@id='state']//option[@value= 'HI']");
```
should write the web element locator in separate line then call into a method/function

chiennv2042 commented on demo/src/test/java/com/example/test/pages/KBBMOOCProject/WhatMyCarWorthPage/OptionPage.java:
```diff
@@ -0,0 +1,31 @@
+    }
+
+    public boolean compareCarStyleAndCategoryNameIsSimilarToStyleAnCarDataInFileDatas(String categoryandstylenameinoption) {
+        WebElement xpathCategoryAndStyle = keyword.findWebElementByXpath("//div[@class='css-1itv5e3 eisth834']/p");
```
should write the web element locator in separate line then call into a method/function

chiennv2042 commented on demo/src/test/java/com/example/test/tests/KBBMOOCProject/KbbBaseTest.java:
```diff
@@ -11,7 +11,7 @@ public class KbbBaseTest {
     public WebDriver driver;
     public CustomKeyword keyword;
-    public String pathChromeDriver = "src\\test\\java\\com\\example\\test\\core\\drivers\\driver\\chromedriver.exe";
+    public String pathChromeDriver = "src//test//java//com//example//test//resources//driver//chromedriver";
```
should using the most popular root path of file

chiennv2042 reviewed (changes requested):
please make your PR have change as the comment inline then make me know to review again

chiennv2042 commented on demo/src/test/java/com/example/test/pages/KBBMOOCProject/WhatMyCarWorthPage/CategoryAndStylePage.java:
```diff
@@ -0,0 +1,54 @@
+    }
+
+    public void clickOnCoupeOptionInCategorySectionInCategoryAndStylePage() throws InterruptedException {
+        WebElement btnCoupe = keyword.findWebElementByXpath("//div[text()='Coupe']");
```
should write the web element locator in separate line then call into a method/function

chiennv2042 commented on demo/src/test/java/com/example/test/pages/KBBMOOCProject/WhatMyCarWorthPage/MyCarValuePage.java:
```diff
@@ -0,0 +1,86 @@
+      // TODO Auto-generated constructor stub
+   }
+
+   WebElement btnMakeModel = keyword.findWebElementByXpath("//div[text()='Make/Model']");
```
should add access modifier for WebElement

chiennv2042 commented on demo/src/test/java/com/example/test/pages/KBBMOOCProject/WhatMyCarWorthPage/MyCarValuePage.java:
```diff
@@ -0,0 +1,86 @@
+   }
+
+   WebElement btnMakeModel = keyword.findWebElementByXpath("//div[text()='Make/Model']");
+   WebElement btnLicensePlate = keyword.findWebElementByXpath("//div[text()='License Plate']");
```
should add access modifier for WebElement

chiennv2042 reviewed (changes requested):
please make your PR have change as the comment inline then make me know to review again

KRJuuga committed:
Changed Base on Comment

Rename file dataJson.java to dataJsonforStyleAndCategoryPage.java
Rename file datas.json to datasforstyleandcategory.json
Rename url to urlforwhatmycarworth
Change path suitable for window user 
Remove thread sleep and use implicitlyWait
Write web element locator in separate line then call into a method/function

KRJuuga committed:
Delete dataJson.java

KRJuuga committed:
Delete datas.json

KRJuuga committed:
Delete phpaccountdata.json

KRJuuga committed:
Delete demo/src/test/java/com/example/test/tests/KBBMOOCProject/CarForSaleTests directory

KRJuuga committed:
Delete demo/src/test/java/com/example/test/tests/KBBMOOCProject/WhatMyCarWorthTest directory

KRJuuga committed:
Delete demo/src/test/java/com/example/test/pages/KBBMOOCProject/WhatMyCarWorthPage directory

chiennv2042 commented:
After your changes, I think your PR can be merge

KRJuuga committed:
Delete configs.properties

KRJuuga committed:
Delete CustomKeyword.java

KRJuuga committed:
Fix conflict

KRJuuga committed:
Fix conflicts

KRJuuga committed:
Fix Conflict

KRJuuga committed:
Fix COnflict

KRJuuga committed:
Resolve Conflict

KRJuuga committed:
Delete CustomKeywordForCategoryAndStylePage.java

KRJuuga committed:
Resolve Conflict
